src/utils/helper.ts

```diff
@@ -25,27 +25,52 @@ export function generateNftId(tokenAddress: string, tokenId: string): string {
   return tokenAddress + '-' + tokenId
 }
 
-export function generateRoleId(grantor: Account, grantee: Account, nft: Nft, role: Bytes): string {
-  return grantor.id + '-' + grantee.id + '-' + nft.id + '-' + role.toHex()
+export function generateRoleAssignmentId(grantor: Account, grantee: Account, nft: Nft, roleAssignment: Bytes): string {
+  return grantor.id + '-' + grantee.id + '-' + nft.id + '-' + roleAssignment.toHex()
 }
 
-export function findOrCreateRole(event: RoleGranted, grantor: Account, grantee: Account, nft: Nft): Role {
-  const roleId = generateRoleId(grantor, grantee, nft, event.params._role)
-  let role = Role.load(roleId)
+export function findOrCreateRoleAssignment(
+  event: RoleGranted,
+  grantor: Account,
+  grantee: Account,
+  nft: Nft,
+): RoleAssignment {
+  const roleAssignmentId = generateRoleAssignmentId(grantor, grantee, nft, event.params._role)
+  let roleAssignment = RoleAssignment.load(roleAssignmentId)
+
+  if (!roleAssignment) {
+    roleAssignment = new RoleAssignment(roleAssignmentId)
+    roleAssignment.role = findOrCreateRole(nft, event.params._role).id
+    roleAssignment.nft = nft.id
+    roleAssignment.grantor = grantor.id
+    roleAssignment.grantee = grantee.id
+  }
+
+  roleAssignment.expirationDate = event.params._expirationDate
+  roleAssignment.revocable = event.params._revocable
+  roleAssignment.data = event.params._data
+  roleAssignment.timestamp = event.block.timestamp
```

Comment:
Should we modify the `roleAssignment.timestamp`, if `timestamp` is the date when the role assignment was grante, maybe we should maintain the original one set when created